pkg/virt-controller/services/template.go

```diff
@@ -259,12 +260,27 @@ func (t *templateService) GetLauncherImage() string {
 }
 
 func (t *templateService) RenderLaunchManifestNoVm(vmi *v1.VirtualMachineInstance) (*k8sv1.Pod, error) {
-	return t.renderLaunchManifest(vmi, nil, true)
+	backendStoragePVCName := ""
+	if backendstorage.IsBackendStorageNeededForVMI(&vmi.Spec) {
+		backendStoragePVC := backendstorage.PVCForVMI(t.persistentVolumeClaimStore, vmi)
+		backendStoragePVCName = backendStoragePVC.Name
+		if backendStoragePVCName == "" {
+			return nil, fmt.Errorf("can't generate manifest without backend-storage PVC")
+		}
+	}
+	return t.renderLaunchManifest(vmi, nil, backendStoragePVCName, true)
 }
 
-func (t *templateService) RenderMigrationManifest(vmi *v1.VirtualMachineInstance, sourcePod *k8sv1.Pod) (*k8sv1.Pod, error) {
+func (t *templateService) RenderMigrationManifest(vmi *v1.VirtualMachineInstance, migration *v1.VirtualMachineInstanceMigration, sourcePod *k8sv1.Pod) (*k8sv1.Pod, error) {
 	imageIDs := containerdisk.ExtractImageIDsFromSourcePod(vmi, sourcePod)
-	targetPod, err := t.renderLaunchManifest(vmi, imageIDs, false)
+	backendStoragePVCName := ""
+	if backendstorage.IsBackendStorageNeededForVMI(&vmi.Spec) {
```

Comment:
You must also check that `migration != nil` otherwise this will panic everytime migration is a nil pointer.

Reply:
I don't think migration can ever be nil, it's passed around to most migration controller functions, none of which ensure it's non-nil...
(I know I break that rule by setting it to nil in unit tests, but it's fine since they don't use backend storage)

Reply:
don't we have unit tests for this?

Reply:
Yeah we're definitely missing at least a unit test.